### PR TITLE
Feature/misc

### PR DESCRIPTION
**Pull request by junjihashimoto**
- Fix a bug of upsampleNearest2d
- Add RawFile type class to access raw file of numpy format.
- Improve a implementation of show tensor
- Add detach function

### CODE REVIEW

**Review by tscholak — APPROVED** [review]
ok, these are nice and useful additions. I didn't check the file IO implementation, a unit test with an actual numpy file would be a great addition :)

**Review by tscholak — APPROVED** [review]
LGTM!

**Comment by junjihashimoto** [comment]
@tscholak 
Thank you for your reviewing!

**Comment by tscholak** [comment]
No problem, nice work!

File: hasktorch/test/SerializeSpec.hs [hunk 1]
```diff
@@ -20,3 +22,20 @@ spec = do
...
+    -- torch.tensor([1,2,3,4], dtype=torch.float32).numpy().tofile(f)
+    -- 
+    new <- System.IO.withFile "test/data/numpy_rawfile" System.IO.ReadMode $
+             \h -> loadBinary h org
+    (asValue new :: [Float]) `shouldBe` [1,2,3,4]
+    System.IO.withFile "numpy_rawfile" System.IO.WriteMode $
+      \h -> saveBinary h new
+    new' <- System.IO.withFile "numpy_rawfile" System.IO.ReadMode $
+             \h -> loadBinary h org
+    (asValue new' :: [Float]) `shouldBe` [1,2,3,4]
```
Comment (junjihashimoto) [inline hasktorch/test/SerializeSpec.hs:41]: The test of serialization is here.
